src/main/java/games/strategy/triplea/attachments/AbstractRulesAttachment.java

```diff
@@ -75,9 +77,9 @@ public void setPlayers(final ArrayList<PlayerID> value) {
     m_players = value;
   }
 
-  public ArrayList<PlayerID> getPlayers() {
+  public List<PlayerID> getPlayers() {
     if (m_players.isEmpty()) {
-      return new ArrayList<>(Collections.singletonList((PlayerID) getAttachedTo()));
+      return Arrays.asList((PlayerID) getAttachedTo());
```

Comment:
Note that there is a change in runtime behavior here.  Previously, the caller, if they wanted to, could call methods like `add()` and `remove()` on the returned list.  Now that it's an array-backed list, those operations will fail with an `UnsupportedOperationException`.

Since the other branch always returns a fully-mutable list, we probably want to be consistent on this branch, as well.

Reply:
You are indeen correct, but after analyzing the code I could confirm this collection is only being used to iterate over

Reply:
However it seems this patters in used in the othert attachment classes as well..
Reverting in this case.